### PR TITLE
[CDAP-5723] fix pipeline name validation to allow hyphen

### PR DESCRIPTION
**Pull request by elfenheart**

### CODE REVIEW

**Comment by ajainarayanan** [comment]
👍 LGTM

File: cdap-ui/app/services/constants.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ angular.module(PKG.name + '.services')
             'MISSING-REQUIRED-FIELDS': 'is missing required fields',
             'NO-SOURCE-FOUND': 'Please add a source to your pipeline',
             'MISSING-NAME': 'Pipeline name is missing.',
-            'INVALID-NAME': 'Pipeline names can only contain alphanumeric (\'a-z A-Z 0-9\') and underscore ( \'_\') characters. Please remove any other characters.',
+            'INVALID-NAME': 'Pipeline names can only contain alphanumeric (\'a-z A-Z 0-9\'), underscore (\'_\'), and hyphen (\'-\') characters. Please remove any other characters.',
```
Comment (anwar6953) [inline cdap-ui/app/services/constants.js]: Do we actually support underscore?

File: cdap-ui/app/features/hydratorplusplus/services/non-store-pipeline-error-factory.js [hunk 2]
```diff
@@ -126,7 +126,7 @@ let isNodeNameUnique = (myHelpers, nodeName, nodes, cb) => {
 
 let hasValidName = (name, cb) => {
   let error;
-  let pattern = /^[\w]+$/;
+  let pattern = /^[\w-]+$/;
```
Comment (anwar6953) [inline cdap-ui/app/features/hydratorplusplus/services/non-store-pipeline-error-factory.js]: This pattern doesn't seem like it supports underscore.

File: cdap-ui/app/services/constants.js (hunk 1 above)
Comment (ajainarayanan) [inline cdap-ui/app/services/constants.js]: Yes I believe so. @elfenheart just created a pipeline with names just having underscores and hyphens. :)

File: cdap-ui/app/features/hydratorplusplus/services/non-store-pipeline-error-factory.js (hunk 2 above)
Comment (anwar6953) [inline cdap-ui/app/features/hydratorplusplus/services/non-store-pipeline-error-factory.js]: Hmm, nevermind.
http://www.w3schools.com/jsref/jsref_regexp_wordchar.asp
"A word character is a character from a-z, A-Z, 0-9, including the _ (underscore) character."

**Comment by anwar6953** [comment]
LGTM

**Comment by elfenheart** [comment]
demoed to @sreevatsanraman

**Pull request merged**